lib/mix/lib/mix/tasks/format.ex

```diff
@@ -298,7 +303,21 @@ defmodule Mix.Tasks.Format do
     sigils =
       for plugin <- plugins,
           sigil <- find_sigils_from_plugins(plugin, formatter_opts),
-          do: {sigil, &plugin.format(&1, &2 ++ formatter_opts)}
+          do: {sigil, plugin}
+
+    sigils =
+      Keyword.keys(sigils)
+      |> Enum.uniq()
+      |> Enum.map(fn sigil ->
+        plugins = Keyword.get_values(sigils, sigil)
+
+        {sigil,
+         fn input, opts ->
+           Enum.reduce(plugins, input, fn plugin, input ->
+             plugin.format(input, opts ++ formatter_opts)
+           end)
+         end}
+      end)
```

Comment:
Maybe a group_by makes more sense?

    sigils = Enum.group_by(sigils) |> Enum.map(fn {sigil, plugins} -> ... end)

Reply:
Makes a lot more sense

For the `Enum.group_by` functions I wasn't sure if there was something more clear than
```elixir
&elem(&1, 0), &elem(&1, 1)
```
 to grab the `sigil` and `plugin`